src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java

```diff
@@ -421,7 +421,8 @@ private static void addLocalVariables(DetailAST varDefAst, Deque<VariableDesc> v
         final DetailAST parentAst = varDefAst.getParent();
         final DetailAST grandParent = parentAst.getParent();
         final boolean isInstanceVarInAnonymousInnerClass =
-                grandParent.getType() == TokenTypes.LITERAL_NEW;
+                grandParent.getType() == TokenTypes.LITERAL_NEW
+                || grandParent.getType() == TokenTypes.CLASS_DEF;
```

Comment:
`|| grandParent.getType() == TokenTypes.CLASS_DEF;` this will match inner classes that is not AnonymousInnerClass

you probable need to not change this variable and extend 
```
if (isInstanceVarInAnonymousInnerClass
                || grandParent.getType() == TokenTypes.CLASS_DEF
                || parentAst.getType() != TokenTypes.OBJBLOCK) {
```

or better
```
        final boolean isInstanceVarInInnerClass =
                grandParent.getType() == TokenTypes.LITERAL_NEW
                || grandParent.getType() == TokenTypes.CLASS_DEF;
        if (isInstanceVarInInnerClass
                || parentAst.getType() != TokenTypes.OBJBLOCK) {
```

Reply:
done